var/spack/repos/builtin/packages/hypre/package.py

```diff
@@ -56,6 +57,8 @@ class Hypre(Package):
     variant('openmp', default=False, description='Enable OpenMP support')
     variant('debug', default=False,
             description='Build debug instead of optimized version')
+    variant('blas', default=True, description='Use external BLAS library')
+    variant('lapack', default=True, description='Use external LAPACK library')
```

Comment:
If these are disabled, does hypre build with internal copies of blas/lapack? I don’t think we want that, as we could end up with different blas/lapack libraries in the DAG.

Reply:
Correct, it'll build with internal copies of blas/lapack. Sorry, I forgot about the higher-level philosophy of "only one instance in the DAG".

For background: this proposed patch comes from an internal package repo we've been maintaining. We're struggling with getting a long-lived codebase and all of its dependencies onto a completely Spack-managed tree; these two variants let us get from "hand-built hypre 2.9.0b with internal BLAS/LAPACK routines" to "Spack-managed hypre 2.9.0b with internal BLAS/LAPACK routines", on the way to "Spack-managed current hypre with Spack-managed BLAS/LAPACK libraries".

If these kind of variants aren't consistent with Spack's larger philosophy, I can decline this PR and hold onto these changes internally until we can take the larger step to "newer hypre + external BLAS/LAPACK".

Reply:
> If these kind of variants aren't consistent with Spack's larger philosophy, I can decline this PR and hold onto these changes internally until we can take the larger step to "newer hypre + external BLAS/LAPACK".

I think that's probably the best plan. There are certainly some packages that build with internal vendored copies of software, but I'm trying to change that. Spack is a package manager after all, it works best when it can manage every dependency in the DAG.

Reply:
OK, thanks for the feedback.